cmd/hatchet-engine/engine/run.go

```diff
@@ -878,6 +888,75 @@ func runV1Config(ctx context.Context, sc *server.ServerConfig) ([]Teardown, erro
 		})
 	}
 
+	var localScheduler *schedulerv1.Scheduler
+
+	if sc.HasService("all") || sc.HasService("scheduler") {
+		partitionCleanup, err := p.StartSchedulerPartition(ctx)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not create create scheduler partition: %w", err)
+		}
+
+		teardown = append(teardown, Teardown{
+			Name: "scheduler partition",
+			Fn:   partitionCleanup,
+		})
+
+		// create the dispatcher
+		s, err := scheduler.New(
+			scheduler.WithAlerter(sc.Alerter),
+			scheduler.WithMessageQueue(sc.MessageQueue),
+			scheduler.WithRepository(sc.EngineRepository),
+			scheduler.WithLogger(sc.Logger),
+			scheduler.WithPartition(p),
+			scheduler.WithQueueLoggerConfig(&sc.AdditionalLoggers.Queue),
+			scheduler.WithSchedulerPool(sc.SchedulingPool),
+		)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not create dispatcher: %w", err)
+		}
+
+		cleanup, err := s.Start()
+
+		if err != nil {
+			return nil, fmt.Errorf("could not start dispatcher: %w", err)
+		}
+
+		teardown = append(teardown, Teardown{
+			Name: "scheduler",
+			Fn:   cleanup,
+		})
+
+		sv1, err := schedulerv1.New(
+			schedulerv1.WithAlerter(sc.Alerter),
+			schedulerv1.WithMessageQueue(sc.MessageQueueV1),
+			schedulerv1.WithRepository(sc.EngineRepository),
+			schedulerv1.WithV2Repository(sc.V1),
+			schedulerv1.WithLogger(sc.Logger),
+			schedulerv1.WithPartition(p),
+			schedulerv1.WithQueueLoggerConfig(&sc.AdditionalLoggers.Queue),
+			schedulerv1.WithSchedulerPool(sc.SchedulingPoolV1),
+		)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not create scheduler (v1): %w", err)
+		}
+
+		cleanup, err = sv1.Start()
+
+		if err != nil {
+			return nil, fmt.Errorf("could not start scheduler (v1): %w", err)
+		}
+
+		teardown = append(teardown, Teardown{
+			Name: "schedulerv1",
+			Fn:   cleanup,
+		})
+
+		localScheduler = sv1
+	}
```

Comment:
Just moved this up and added the `localScheduler` assignment, but otherwise no code changes.